### PR TITLE
Move WithAzureArmTemplate* and WithAwsCfTemplate* annotations

### LINKED ISSUE #120: Move vendor specific annotations to designated modules
Annotations like `@WithAzureArmTemplate` and `@WithAwsCfTemplate` should be placed in the modules.

### PR DESCRIPTION
**Pull request by istraka**
Fixes #120 

FYI @mchoma

### CODE REVIEW

File: aws/src/main/java/aws/core/AwsSunstoneDeployer.java [hunk 1]
```diff
@@ -23,11 +28,21 @@
...
+            deployCfTemplate((WithAwsCfTemplate) annotation, store);
+        } else if (WithAwsCfTemplateRepeatable.class.isAssignableFrom(annotation.annotationType())) {
+            for (WithAwsCfTemplate withAwsCfTemplate : ((WithAwsCfTemplateRepeatable) annotation).value()) {
+                deployCfTemplate(withAwsCfTemplate, store);
+            }
+        }
+    }
+
+    private void deployCfTemplate(WithAwsCfTemplate awsTemplateDefinition, AwsSunstoneStore store) throws
+        SunstoneException {
```
Comment (mnovak1) [inline aws/src/main/java/aws/core/AwsSunstoneDeployer.java:45]: `SunstoneException` can be removed.

**Review by mnovak1 — COMMENTED** [review]
Just one comment. Otherwise LGTM

File: aws/src/main/java/aws/core/AwsSunstoneDeployer.java (hunk 1 above)
Comment (istraka) [reply aws/src/main/java/aws/core/AwsSunstoneDeployer.java:45]: Oh, I didn't mean to throw unchecked exceptions. I have changed those to Sunstone exceptions. See the amended commit please.

**Review by mnovak1 — APPROVED** [review]

**Review by mchoma — COMMENTED** [review]
This is aligned to what we talked before. Approving.

**Pull request merged**